relay-spans/src/span.rs

```diff
@@ -208,7 +208,7 @@ pub fn otel_to_sentry_span(otel_span: OtelSpan) -> EventSpan {
             grpc_status_code,
         )),
         timestamp: Timestamp(end_timestamp).into(),
-        trace_id: TraceId(trace_id).into(),
+        trace_id: TraceId::parse_str_annotated(&trace_id),
```

Comment:
I decided to implement annotated in case of errors instead of using `Result`, so that we can have in the payload the errors.

My only worry is what will happen upstream in case no `trace_id` is set.